indexing-service/src/test/java/io/druid/indexing/overlord/TaskLifecycleTest.java

```diff
@@ -172,94 +222,99 @@ private static InputRow IR(String dt, String dim1, String dim2, float met)
     );
   }
 
-  private static FirehoseFactory newMockExceptionalFirehoseFactory()
+  private static class MockExceptionalFirehoseFactory implements FirehoseFactory
```

Comment:
Also, this isn't really a mock. Perhaps ExceptionThrowingFirehoseFactory can be the name?